File: 61-rotate-list/rotate-list.ts
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     val: number
 *     next: ListNode | null
 *     constructor(val?: number, next?: ListNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.next = (next===undefined ? null : next)
 *     }
 * }
 */

function rotateRight(head: ListNode | null, k: number): ListNode | null {

  let ans = head 
  let res = head
  let n = 0
if (!head) return null

  while (head){
  n += 1 
  head = head.next
  }

     k = k % n
  if (k === 0) return ans
    const reach = n - k
    let poi = 0
    let start = null
    let curr = ans  
while (curr){
    poi += 1
    if (poi == reach){
        start = curr.next
        curr.next = null
        break
    } else {
        curr = curr.next
    }
}let fin_ans = start
        let prev = null
    while(start){

        prev = start
        start = start.next
    }
    prev.next = res



  return fin_ans
};

File: 61-rotate-list/rotate-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def rotateRight(self, head: Optional[ListNode], k: int) -> Optional[ListNode]:
        n=head
        d=head
        g=head
        l=[]
        if head is None:return
        while n:
            l.append(n.val)
            n=n.next
        n=len(l)
        
        if k>n:
            
            k=k%n
            
        print(k)    
        l3=l[:n-k]
        l2=l[n-k:(n-k)+k]
        # print(l2)
        # l2=l2[::-1]
        l1=l2+l3 
        # print(l2)
        # print(l3)
        i=0
        while d:
            d.val=l1[i]
            d=d.next
            i+=1
        return g